sdk/internal/recording/recording.go

```diff
@@ -473,11 +473,35 @@ func defaultOptions() *RecordingOptions {
 	}
 }
 
-func (r RecordingOptions) hostScheme() string {
+func (r RecordingOptions) RouteURL(t *testing.T, rawReq *http.Request) {
+	if GetRecordMode() != LiveMode && !IsLiveOnly(t) {
+		originalURLHost := rawReq.URL.Host
+		rawReq.URL.Scheme = r.scheme()
+		rawReq.URL.Host = r.host()
+		rawReq.Host = r.host()
+
+		rawReq.Header.Set(UpstreamURIHeader, fmt.Sprintf("%v://%v", r.scheme(), originalURLHost))
+		rawReq.Header.Set(ModeHeader, GetRecordMode())
+		rawReq.Header.Set(IDHeader, GetRecordingId(t))
+	}
+}
+
+func (r RecordingOptions) host() string {
+	if r.UseHTTPS {
+		return "localhost:5001"
+	}
+	return "localhost:5000"
+}
+
+func (r RecordingOptions) scheme() string {
 	if r.UseHTTPS {
-		return "https://localhost:5001"
+		return "https"
 	}
-	return "http://localhost:5000"
+	return "http"
+}
+
+func (r RecordingOptions) hostScheme() string {
```

Comment:
Nitpicking: this is often called the "base URL" in practice--everything before the path. I don't know that there's a proper term for it in the URL schema.